HouseRules_Essentials/Rules/StatModifiersOverridenRule.cs

```diff
@@ -37,15 +38,23 @@ protected override void OnDeactivate(GameContext gameContext)
             ReplaceStatModifiers(_originals);
         }
 
-        private static Dictionary<string, int> ReplaceStatModifiers(Dictionary<string, int> replacements)
+        private static Dictionary<AbilityKey, int> ReplaceStatModifiers(Dictionary<AbilityKey, int> replacements)
         {
-            var originals = new Dictionary<string, int>();
+            var originals = new Dictionary<AbilityKey, int>();
 
             var statModifierType = AccessTools.TypeByName("Boardgame.GameplayEffects.StatModifier");
-            var statModifiers = Resources.FindObjectsOfTypeAll(statModifierType);
             foreach (var replacement in replacements)
             {
-                var statModifier = statModifiers.First(c => c.name.Equals($"{replacement.Key}(Clone)"));
+                if (!AbilityFactory.TryGetAbility(replacement.Key, out var ability))
+                {
+                    throw new InvalidOperationException($"AbilityKey [{replacement.Key}] does not have a corresponding ability.");
+                }
+
+                if (!ability.TryGetComponent(statModifierType, out var statModifier))
+                {
+                    throw new InvalidOperationException($"AbilityKey [{replacement.Key}] does not have a corresponding StatModifier.");
```

Comment:
As we're doing `TryGetComponent` on a `statModifierType`, shouldn't this exception say `StatModifier` rather than `AbilityKey`